lib/repl.js

```diff
@@ -1911,6 +1922,33 @@ function includesProxiesOrGetters(expr, exprStr, evalFn, ctx, callback) {
   return callback(false);
 }
 
+function reclusiveCatchPromise(obj, seen = new SafeWeakSet()) {
```

Comment:
This reclusive code is necessary to handle errors that might be thrown from within objects or similar structures.
https://github.com/nodejs/node/pull/58943/files#diff-24ded3e78d8bb56bd21deb058a4766995ee891242cf993a9a185c57213ce38c0R15

Reply:
I don't think we should always iterate through the object here at all. We already do that with the AST and having a separate way to iterate it again does not sound correct to me.

I also believe we could prevent reading the property, if we check in the AST what the property is like and just stop iterating, if we find it's a function call.